media_service/tests/test_models.py

```diff
@@ -49,6 +50,36 @@ def test_get_image(self):
             self.assertEqual(thumb_actual['height'], thumb_h)
             self.assertTrue(thumb_actual['url'])
 
+class TestResource(unittest.TestCase):
+    test_course = None
+    
+    @classmethod
+    def setUpClass(cls):
+        cls.test_course = Course(title="Test Course" , lti_context_id="ABC123", lti_tool_consumer_instance_guid="canvas")
+        cls.test_course.save()
+
+    def test_save_with_metadata(self):
+        title = "Test Resource"
+        course = TestResource.test_course
+        default_metadata = metadata_default()
+        missing_metadata = ["", None]
+        invalid_metadata = [json.dumps(v) for v in None, True, False, 123, {}]
```

Comment:
If you used a json-validation approach, instead of just checking that it's a string that looks like it has a list at the top level, then invalid metadata could include some malformed json.